sklearn/metrics/_pairwise_distances_reduction/_argkmin.pyx.tp

```diff
@@ -98,14 +98,8 @@ cdef class ArgKmin{{name_suffix}}(BaseDistanceReducer{{name_suffix}}):
                 metric_kwargs=metric_kwargs,
             )
         else:
-            # Fall back on a generic implementation that handles all distance
-            # metrics by computing it between 2 vectors at a time.
-
-            # The extra `Y_norm_squared` argument for the back-end is only
-            # supported for the Euclidean variant.
-            if metric_kwargs is not None:
-                metric_kwargs.pop("Y_norm_squared", None)
-
```

Comment:
Maybe we should instead change the upstream estimator objects to avoid passing precomputed `Y_norm_squared` when the train or test data is sparse.

If it's too complicated, then I am fine with keeping the `pop` trick.

Reply:
Resolving this conversation as we agreed on a solution in #24272.